metrics/sinks/influxdb/influxdb_test.go

```diff
@@ -164,3 +170,323 @@ func TestCreateInfluxdbSink(t *testing.T) {
 	//check sink name
 	assert.Equal(t, sink.Name(), "InfluxDB Sink")
 }
+
+func makeRow(results [][]string) influxql.Row {
+	resRow := influxql.Row{
+		Values: make([][]interface{}, len(results)),
+	}
+
+	for setInd, valSet := range results {
+		outVals := make([]interface{}, len(valSet))
+		for valInd, val := range valSet {
+			if valInd == 0 {
+				// timestamp should just be a string
+				outVals[valInd] = val
+			} else {
+				outVals[valInd] = json.Number(val)
+			}
+		}
+		resRow.Values[setInd] = outVals
+	}
+
+	return resRow
+}
+
+func checkMetricVal(expected, actual core.MetricValue) bool {
+	if expected.ValueType != actual.ValueType {
+		return false
+	}
+
+	// only check the relevant value type
+	switch expected.ValueType {
+	case core.ValueFloat:
+		return expected.FloatValue == actual.FloatValue
+	case core.ValueInt64:
+		return expected.IntValue == actual.IntValue
+	default:
+		return expected == actual
+	}
+}
+
+func TestHistoricalInfluxRawMetricsParsing(t *testing.T) {
+	// in order to just test the parsing, we just go directly to the sink type
```

Comment:
This tests turning the JSON returned by InfluxDB for a  "query row" (which is actually a set of results) into internal Heapster form (both for the normal queries and the aggregated queries).